src/Root.cpp

```diff
@@ -451,10 +451,10 @@ void cRoot::LoadWorlds(cSettingsRepositoryInterface & a_Settings, bool a_IsNewIn
 			}
 			NewWorld = new cWorld(WorldName.c_str(), dimNether, LinkTo);
 		}
-		// if the world is called x_end
+		// if the world is called x_the_end
 		else if ((LowercaseName.size() > EndAppend.size()) && (LowercaseName.substr(LowercaseName.size() - EndAppend.size()) == EndAppend))
```

Comment:
The problem here is that if a player removes their existing `world_end`, it would be recreated as an overworld called `world_end`. I think this code block should catch both `_end` and `_the_end`. In other words, both prefixes are to be allowed, but `_the_end` is the default.